authors/apps/profiles/test/test_user_profile.py

```diff
@@ -0,0 +1,52 @@
+import json
+from django.urls import reverse
+from rest_framework.views import status
+from rest_framework.test import APITestCase, APIClient
+
+
+class LoginTestCase(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.login_url = reverse('authentication:auth-login')
+        self.signup_url = reverse('authentication:auth-register')
+        self.signup_data = {
+            "user": {
+                "username": "johndoe",
+                "email": "john@gmail.com",
+                "password": "@Qwerty12345"
+                }}
+        self.profile_data = {
+            "first_name":"",
+            "last_name":"",
+            "gender":"M",
+            "bio":"",
+            "profile_image":None
+            }
+
+    def test_get_profile(self):
+        """Test get profile upon registrations"""
+        # register user
+        register = self.client.post(self.signup_url,self.signup_data,format='json')
+        self.assertEqual(register.status_code, status.HTTP_201_CREATED)
+        
+        # get user profile
+        profile = self.client.get(reverse('profiles:get-profile', kwargs={'slug': 'johndoe'}),format='json')
+        self.assertEqual(self.profile_data,profile.data)
+    
+    def test_post_profile(self):
+        """Test post profile upon registrations"""
+        # override self.profile_data 
+        self.profile_data['first_name'],self.profile_data['last_name'] = 'kwame', 'asiago'
+        self.profile_data['bio'] = 'some bio'
+        del self.profile_data['profile_image']
```

Comment:
This test looks complicated, please simplify it. Consider, creating new variables like the profile_data you need instead of overriding what's already available.